### PR TITLE
test(service): Add e2e tests for min, max scale options

### LINKED ISSUE #254: Add integration tests for min/max scale options of service create and update
For service create and service update, we are missing coverage for following options
```
      --max-scale int            Maximal number of replicas.
      --min-scale int            Minimal number of replicas.
```

### PR DESCRIPTION
**Pull request by navidshaikh**
/lint

Fixes #254 

## Proposed Changes
* add e2e tests for min, max scale options for `service create` and `service update` operations.

### CODE REVIEW

File: test/e2e/service_options_test.go [hunk 1]
```diff
@@ -49,6 +49,30 @@ func TestServiceOptions(t *testing.T) {
 	t.Run("delete hello service and returns no error", func(t *testing.T) {
 		test.serviceDelete(t, "hello")
 	})
+
+	t.Run("create hello service with min/max scale options and returns no error", func(t *testing.T) {
+		test.serviceCreateWithOptions(t, "hello", []string{"--min-scale", "1", "--max-scale", "3"})
+	})
+
+	t.Run("validate min scale value for service", func(t *testing.T) {
+		test.validateServiceMinScale(t, "hello", "1")
```
Comment (rhuss) [inline test/e2e/service_options_test.go]: I would add the validation step after the service call itself but in the same subtest, to make it not that fine granular.

**Comment by rhuss** [comment]
@navidshaikh I highly suspect that the inegration test error is because the `kn delete` right before that step is async and the route as dependenct object is not yet deleted when you create a new service. 

For now, I'd suggest to just rename the service from "hello" to something like "hello-2"

**Comment by navidshaikh** [comment]
@rhuss : Thanks Roland, I've renamed the service.

File: test/e2e/service_options_test.go (hunk 1 above)
Comment (navidshaikh) [reply test/e2e/service_options_test.go]: fixed it, thanks!

**Comment by navidshaikh** [comment]
Failure of `pull-knative-client-integration-tests-latest-release` job was because it was returning the deprecated fields while we were after new fields https://github.com/knative/client/pull/302#issuecomment-515297563.

Updated PR to fallback to work with old and new fields for jsonpath lookup.

**Comment by rhuss** [comment]
@googlebot rescan

**Pull request merged**